torch/onnx/symbolic_opset9.py

```diff
@@ -3483,19 +3466,21 @@ def tensor(
     dtype = symbolic_helper._get_const(dtype, "i", "dtype")
     if symbolic_helper._is_packed_list(data):
         if dtype is None:
-            scalar_name = symbolic_helper._unpack_list(data)[0].type().scalarType()
-            dtype = _type_utils.JitScalarType.from_name(scalar_name)
+            dtype = _type_utils.JitScalarType.from_value(
+                symbolic_helper._unpack_list(data)[0]
+            ).scalar_name()
         input_list = list()
         for t in symbolic_helper._unpack_list(data):
             shape_reference = g.op("Constant", value_t=torch.LongTensor([1]))
             t = symbolic_helper._reshape_helper(g, t, shape_reference)
-            t = g.op("Cast", t, to_i=_type_utils.JitScalarType(dtype).onnx_type())
+            t = g.op(
+                "Cast", t, to_i=_type_utils.JitScalarType.from_name(dtype).onnx_type()
+            )
```

Comment:
Remove the call to `.scalar_name()` [line 3471] and do `dtype.onnx_type()` on line 3477? Possibly rename `dtype` to `xxx_scalar_type` as well to reflect the content of the var on 3469

Reply:
Cast", t, to_i